src/future.rs

```diff
@@ -26,20 +26,43 @@ impl <COMP: Component> ComponentLinkFuture for ComponentLink<COMP> {
         let mut link: ComponentLink<COMP>  = self.clone();
         let js_future = async move{
             let message: COMP::Message = future.await;
-            link.send_self(message);
-            Ok(JsValue::NULL)
+            link.send_message(message);
         };
-        future_to_promise(js_future);
+        spawn_local(js_future);
 
     }
 
     fn send_future_batch<F>(&self, future: F) where F: Future<Output=Vec<Self::Message>> + 'static {
         let mut link: ComponentLink<COMP> = self.clone();
         let js_future = async move {
             let messages: Vec<COMP::Message> = future.await;
-            link.send_self_batch(messages);
-            Ok(JsValue::NULL)
+            link.send_message_batch(messages);
+        };
+        spawn_local(js_future);
+    }
+}
+
+/// Trait that allows you to use `AgentLink`s to register futures.
+pub trait AgentLinkFuture {
```

Comment:
Actually, I think we can use only one trait here, called LinkFuture. Implement it both for ComponentLink, and AgentLink. 
Because this is a utility crate and some awkwardness is acceptable for the moment, its fine to have the AgentLink implementation crash via an `unimplemented!("Agents don't support batching their messages.")` for the `send_future_batch()` method.

Reply:
Since Agent will never get support for batching messages. I think it makes sense to have `LinkFuture` and `LinkFutureBatch` as traits where they have the respective methods `send_future` and `send_future_batch`.
`LinkFuture` would be implemented for both `ComponentLink` and `AgentLink`, while `LinkFutureBatch` would only be implemented for `ComponentLink`.

Reply:
Perfect I made the change :)

Reply:
All good then. I might personally split this trait per my second comment here, but for now, this is good enough to merge.